addons/website_forum_doc/controllers/main.py

```diff
@@ -40,31 +36,20 @@ def post(self, toc, post, **kwargs):
 
     @http.route('/forum/<model("forum.forum"):forum>/question/<model("forum.post"):post>/promote', type='http', auth="user", website=True)
     def post_toc(self, forum, post, **kwargs):
-        cr, uid, context, toc_id = request.cr, request.uid, request.context, False
-        user = request.registry['res.users'].browse(cr, uid, uid, context=context)
-        assert user.karma >= 200, 'You need 200 karma to promote a post to the documentation'
-        toc_obj = request.registry['forum.documentation.toc']
-        obj_ids = toc_obj.search(cr, uid, [], context=context)
-        tocs = toc_obj.browse(cr, uid, obj_ids, context=context)
+        request.env.user.karma >= 200, 'You need 200 karma to promote a post to the documentation'
+        tocs = request.env['forum.documentation.toc'].search([]).filtered(lambda toc: not toc.child_ids)
```

Comment:
instead of search for all records , search with not child_ids. then filtered is not required